test/lib/ufe/testConnections.py

```diff
@@ -571,6 +675,135 @@ def testCreateStandardSurface(self):
+
+        # Create a ContextOps interface for the proxy shape.
+        proxyPathSegment = mayaUtils.createUfePathSegment(proxyShape)
+        proxyShapePath = ufe.Path([proxyPathSegment])
+        proxyShapeItem = ufe.Hierarchy.createItem(proxyShapePath)
+        contextOps = ufe.ContextOps.contextOps(proxyShapeItem)
+
+        cmd = contextOps.doOp(['Add New Prim', 'Capsule'])
+        cmd = contextOps.doOp(['Add New Prim', 'Material'])
+
+        rootHier = ufe.Hierarchy.hierarchy(proxyShapeItem)
+
+        materialItem = rootHier.children()[-1]
+        materialAttrs = ufe.Attributes.attributes(materialItem)
+        contextOps = ufe.ContextOps.contextOps(materialItem)
+
+        cmd = contextOps.doOp(['Add New Prim', 'Shader'])
+
+        materialHier = ufe.Hierarchy.hierarchy(materialItem)
+        materialPrim = usdUtils.getPrimFromSceneItem(materialItem)
+
+
+        shaderItem = materialHier.children()[0]
+        shaderAttrs = ufe.Attributes.attributes(shaderItem)
+        shaderPrim = usdUtils.getPrimFromSceneItem(shaderItem)
+
+        shaderAttr = shaderAttrs.attribute("info:id")
+        shaderAttr.set("ND_standard_surface_surfaceshader")
+
+        # The native type of the output has changed in recent versions of USD, so we need to
+        # check with Sdr to see what native type we are going to get.
+        ssNodeDef = Sdr.Registry().GetShaderNodeByIdentifier("ND_standard_surface_surfaceshader")
+        ssOutput = ssNodeDef.GetShaderOutput("out")
+        ssOutputType = ssOutput.GetType()
+
+        #
+        #
+        # Then switch to connection code to connect the shader. Since we never created the
+        # attributes, we expect the connection code to create them.
+        #
+        #
+        shaderOutput = shaderAttrs.attribute("outputs:out")
+        materialOutput = materialAttrs.attribute("outputs:surface")
+
+        self.assertEqual(shaderOutput.type, "Generic")
+        self.assertEqual(shaderOutput.nativeType(), ssOutputType)
+        self.assertEqual(materialOutput.type, "Generic")
+        self.assertEqual(materialOutput.nativeType(), "TfToken")
+
+        connectionHandler = ufe.RunTimeMgr.instance().connectionHandler(materialItem.runTimeId())
+
+        # The attributes are not created yet:
+        self.assertEqual(len(materialPrim.GetAuthoredProperties()), 0)
+        self.assertEqual(len(shaderPrim.GetAuthoredProperties()), 1)
+        self.assertEqual("info:id", shaderPrim.GetAuthoredProperties()[0].GetName())
+
+        connectionCommand = connectionHandler.createConnectionCmd(shaderOutput, materialOutput)
+        connectionCommand.execute()
+
+        def assertNewConnectionMatchesCommand(self, materialItem):
+            connections = connectionHandler.sourceConnections(materialItem)
+            self.assertIsNotNone(connections)
+            conns = connections.allConnections()
+            self.assertEqual(len(conns), 1)
+            # Check the command result:
```

Comment:
We can now find out about connections being remapped by looking at the command result.